app/src/main/java/woowacourse/shopping/data/mapper/CartMapper.kt

```diff
@@ -0,0 +1,27 @@
+package woowacourse.shopping.data.mapper
+
+import woowacourse.shopping.data.CartProductEntity
+import woowacourse.shopping.model.CartProduct
+import woowacourse.shopping.model.Product
+
+fun Product.toEntity(): CartProductEntity {
+    return CartProductEntity(
+        name = name,
+        price = price,
+        imageUrl = imageUrl,
+    )
+}
+
+fun List<CartProductEntity>.toDomain(): List<CartProduct> {
+    return map {
+        CartProduct(
+            product = Product(
+                name = it.name,
+                price = it.price,
+                imageUrl = it.imageUrl,
+            ),
+            id = it.id,
+            createdAt = it.createdAt,
+        )
+    }
+}
```

Comment:
```suggestion
fun List<CartProductEntity>.toDomain(): List<CartProduct> {
    return map { cartProductEntity ->
        CartProduct(
            product = Product(
                name = cartProductEntity.name,
                price = cartProductEntity.price,
                imageUrl = cartProductEntity.imageUrl,
            ),
            id = cartProductEntity.id,
            createdAt = cartProductEntity.createdAt,
        )
    }
}
```
---

조금만 봐도 `it`이 뭔지 유추할 수는 있지만, 지금과 같이 여러 인덴트가 들어간다면 더나은 가독성을 위해 `cartProductEntity`와 같은 **explicit parameter**를 사용하는건 어떨까요?  물론 큰차이가 느껴지지 않는다고 부나가 느끼신다면 무시하셔도 좋습니다.

Reply:
좋은 의견이라고 생각해요~
코드를 보는데 유심히 들여다볼 정도로 복잡하지는 않지만, 명시적으로 작성해주면 좋겠다고 생각했어요.

참고로 저는 사실 대부분의 경우 아래와 같은 기준으로 explict paramter를 사용합니다.
- 하나의 파일에 코드가 길어져서 가독성이 떨어질 때
- 한 함수 내에서 depth가 길어질 때